pdns/dnsname.cc

```diff
@@ -839,6 +839,7 @@ std::string ZoneName::toStringFull(const std::string& separator, const bool trai
     if (d_name.isRoot()) {
       ret.push_back('.');
     }
+    ret.push_back('.');
```

Comment:
I think the `isRoot` check was wrong and that dot should be unconditional: we need to add an explicit dot `if (!trailing)`, then another dot and the variant name.
So the change should be: (and the comment above removed, but the GH UI did not allow me to include that line in the proposed changeset)
```suggestion
    ret.push_back('.');
```